src/compiler/checker.ts

```diff
@@ -10406,14 +10406,34 @@ namespace ts {
             const checkTypeInstantiable = maybeTypeOfKind(checkType, TypeFlags.Instantiable | TypeFlags.GenericMappedType);
             let combinedMapper: TypeMapper | undefined;
             if (root.inferTypeParameters) {
-                const context = createInferenceContext(root.inferTypeParameters, /*signature*/ undefined, InferenceFlags.None);
+                const freshParams = map(root.inferTypeParameters, cloneTypeParameter);
+                const freshMapper = createTypeMapper(root.inferTypeParameters, freshParams);
+                const context = createInferenceContext(freshParams, /*signature*/ undefined, InferenceFlags.None);
+                // We have three mappers that need applying:
+                //    * The original `mapper` used to create this conditional
+                //    * The mapper that maps the old root type parameter to the clone (`freshMapper`)
+                //    * The mapper that maps the clone to its inference result (`context.mapper`)
+                // When we're looking at making an inference for a clone, when we get its constraint, it's autmagically be

```

Comment:
```suggestion
                // When we're looking at making an inference for a clone, when we get its constraint, it'll automagically be
```

Reply:
unrelated, but frustrating: `suggestion`s on `checker.ts` look nice but can't be batched and applied on the web UI for github. F. Where do I go to submit this bug report, I wonder.